src/librustdoc/html/render/search_index.rs

```diff
@@ -315,11 +314,6 @@ fn add_generics_and_bounds_as_types<'tcx>(
         }
     }
 
-    if recurse >= 10 {
```

Comment:
I don't remember if we have a test to ensure it's working. Can you check it please?

Reply:
I don't know what this `recurse` code is for, so can you tell me what kind of test to write?

Reply:
I wanted mostly an investigation on this code line to see in which PR it was added to solve which issue. I \*think\* there is one already, just didn't check yet.

Reply:
Ok, this code was added by #59539, as a "temporary" fix for #59502. It turns out there's no test for it, and the underlying bug is still there: I ran `cargo doc` on the crate and commit described in #59502 using this PR's toolchain and the stack overflow re-appeared.

Reply:
Thanks for looking into it! Please add a test to check this overflow and put back the check.

Reply:
I minimized the issue and added a test.